integration-cli/docker_cli_registry_user_agent_test.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-check/check"
+)
+
+// TestUserAgentPassThroughOnPull verifies that when an image is pulled from
+// a registry, the registry should see a User-Agent string of the form
+// "[client UA] upstream-ua/end; [docker engine UA]"
+func (s *DockerRegistrySuite) TestUserAgentPassThroughOnPull(c *check.C) {
+	reg, err := newTestRegistry(c)
+	c.Assert(err, check.IsNil)
+	expectUpstreamUA := false
+
+	reg.registerHandler("/v2/", func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(404)
+		var ua string
+		for k, v := range r.Header {
+			if k == "User-Agent" {
+				ua = v[0]
+			}
+		}
+		c.Assert(ua, check.Not(check.Equals), "", check.Commentf("No User-Agent found in request"))
+		if r.URL.Path == "/v2/busybox/manifests/latest" {
+			if expectUpstreamUA {
+				c.Assert(ua, check.Matches, ".+upstream\\-ua\\/end;.+", check.Commentf("Seperator token 'upstream-ua/end;' not found"))
```

Comment:
Could this check for the expected client UA? Perhaps with a permissing regexp so the test isn't tied to a particular client version?